### PR TITLE
feat: async migration to move data from unsupported partitions to a backup table (wip)

### PR DESCRIPTION
**Pull request by yakkomajuri**
need to move this data to fix some issues

(sorry short description, going afk)

### CODE REVIEW

**Comment by hazzadous** [comment]
Is there a corresponding change to drop or mark new incoming events with wild timestamps so we don't end up recreating those moved parts on `sharded_events`?

**Comment by yakkomajuri** [comment]
the async migration is generalizable atm, but the corresponding ingestion pr is this https://github.com/PostHog/posthog/pull/16083

File: posthog/async_migrations/migrations/0010_move_old_partitions.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                CREATE TABLE {backup_table_name}
+                ON CLUSTER 'posthog'
+                AS sharded_events
+                ENGINE = ReplicatedReplacingMergeTree('/clickhouse/tables/{shard}/posthog.{backup_table_name}', '{replica}', _timestamp)
+                PARTITION BY toYYYYMM(timestamp)
+                ORDER BY (team_id, toDate(timestamp), event, cityHash64(distinct_id), cityHash64(uuid))
+                SAMPLE BY cityHash64(distinct_id)
+                SETTINGS index_granularity = 8192
+                """,
+                rollback="DROP TABLE IF EXISTS events_backup_0010_move_old_partitions",
```
Comment (hazzadous) [inline posthog/async_migrations/migrations/0010_move_old_partitions.py:69]: Should this be `{backup_table_name}`?

Comment (hazzadous) [reply posthog/async_migrations/migrations/0010_move_old_partitions.py:69]: Also, I guess this will be removing the already moved partitions?

**Review by hazzadous — APPROVED** [review]
Looks good 🤞

File: posthog/async_migrations/migrations/0010_move_old_partitions.py (hunk 1 above)
Comment (yakkomajuri) [reply posthog/async_migrations/migrations/0010_move_old_partitions.py:69]: Hmm yeah that's not great is it - probably don't need a rollback here

Comment (yakkomajuri) [inline posthog/async_migrations/migrations/0010_move_old_partitions.py:69]:
```suggestion
                rollback=None,
```

**Pull request merged**